File: JavaScriptBasics/WhileLoop/Exercise/Walking.js
function Walking(input){

    let target = 10000;
    let index = -1;
    while(target>0){
        index++;
        let command = input[index];
       

        if(command==="Going home"){

            target = target-parseInt(input[index+1]);
            break;
        

        }else{
            target = target-parseInt(input[index]);
        }
    }
    if(target<=0){
        console.log("Goal reached! Good job!");
        console.log(Math.abs(target)+" steps over the goal!")
    }else{
        console.log(target+" more steps to reach goal.")
    }


}

File: JavaBasics/WhileLoops/Exercise/Walking.java
package EntryModule.WhileLoops.Exercise;

import java.util.Scanner;
public class Walking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int target = 10000;
        int build  = 0;
        int meters = 0;

        while(build<target){
            String step =(scanner.nextLine());
            if(step.equals("Going home")){meters = Integer.parseInt(scanner.nextLine());}
            else{meters = Integer.parseInt(step);}
            build = build+meters;
            if(build>=target) {
                System.out.println("Goal reached! Good job!");
                System.out.println((build - target) + " steps over the goal!");
                break;
            }
            if(step.equals("Going home") && build<target){
                System.out.println((target-build)+" more steps to reach goal.");
                break;
            }
            else if(build>=target && step.equals("Going home")) {
                System.out.println("Goal reached! Good job!");
                System.out.println((build - target) + " steps over the goal!");
                break;
            }
        }
    }
}
